### PR TITLE
Feature/sl 982 server search

### PR DESCRIPTION
**Pull request by kpakur**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sandl/snlrules/utils/DateTimeUtils.java [hunk 1]
```diff
@@ -69,4 +70,47 @@ public static String humanizeDate(OffsetDateTime dateTime) {
...
     }
+
+    public static boolean sameDay(OffsetDateTime one, OffsetDateTime two) {
+        OffsetDateTime startTruncated = one.truncatedTo(ChronoUnit.DAYS);
+        OffsetDateTime toCompareStartTruncated = two.truncatedTo(ChronoUnit.DAYS);
+
+        return !startTruncated.isAfter(toCompareStartTruncated) && !startTruncated.isBefore(toCompareStartTruncated);
+    }
+
+    public static boolean contains(OffsetDateTime biggerStart, OffsetDateTime biggerEnd,
```
Comment (leszekgu) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlrules/utils/DateTimeUtils.java]: This method melts my mind a little bit, but I'm not sure if better naming for method o parameters can be applied.
```
(OffsetDateTime containerStart, OffsetDateTimecontainerEnd, OffsetDateTime toContainStart, OffsetDateTime toContainEnd)
```
but then again it doesn't solve a thing.

File: src/test/java/uk/gov/hmcts/reform/sandl/snlrules/rules/DateTimeHelper.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+
+public final class DateTimeHelper {
+    private DateTimeHelper() {
+    }
+
+    public static OffsetDateTime offsetDateTimeOf(String date) {
```
Comment (leszekgu) [inline src/test/java/uk/gov/hmcts/reform/sandl/snlrules/rules/DateTimeHelper.java]: why here as part of a separate class as it's the same as the one in DateTimeUtils.

File: src/main/resources/Search/Search.drl [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      String $judgeId,              // can be null
+      String $roomId,               // can be null
+      Duration $duration,           // mandatory
+      OffsetDateTime $start,        // mandatory
+      OffsetDateTime $end           // mandatory
+      )
+
+      $jb : BookableJudge(
+           judgeId == $judgeId || $judgeId == null,
+           duration >= $duration,
```
Comment (radoslawlandowskikainos) [inline src/main/resources/Search/Search.drl]: Whats the need of explicit check for duration if in lines 122 and 123 you check for starttime and endtime?

Comment (kpakur) [reply src/main/resources/Search/Search.drl]: possibly it may not be needed

**Pull request merged**